ecc_go/chaincode/enclave_go/skvs_stub_interface.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package enclave_go
+
+import (
+	"encoding/json"
+
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/pkg/errors"
+)
+
+const SKVSKey = "SKVS"
+
+type SkvsStubInterface struct {
+	*FpcStubInterface
+	allDataOld map[string][]byte
+	allDataNew map[string][]byte
+	key        string
+}
+
+func NewSkvsStubInterface(stub shim.ChaincodeStubInterface, input *pb.ChaincodeInput, rwset *readWriteSet, sep StateEncryptionFunctions) *SkvsStubInterface {
+	fpcStub := NewFpcStubInterface(stub, input, rwset, sep)
+	skvsStub := &SkvsStubInterface{
+		FpcStubInterface: fpcStub,
+		allDataOld:       map[string][]byte{},
+		allDataNew:       map[string][]byte{},
+		key:              SKVSKey,
+	}
+	err := skvsStub.initSKVS()
+	if err != nil {
+		panic("Initializing SKVS failed")
+	}
+	return skvsStub
+}
+
+func (s *SkvsStubInterface) initSKVS() error {
+
+	// get current state, this will only operate once
+	encValue, err := s.GetPublicState(s.key)
+	if err != nil {
+		return err
+	}
+
+	// return if the key initially does not exist
+	if len(encValue) == 0 {
+		logger.Warningf("SKVS is empty, Initiating.")
+		return nil
+	}
+
+	value, err := s.sep.DecryptState(encValue)
+	if err != nil {
+		return err
+	}
+	logger.Debug("SKVS has default value, loading current value.")
+
+	err = json.Unmarshal(value, &s.allDataOld)
+	if err != nil {
+		logger.Errorf("SKVS Json unmarshal error: %s", err)
+		return err
+	}
+	err = json.Unmarshal(value, &s.allDataNew)
+	if err != nil {
+		logger.Errorf("SKVS Json unmarshal error: %s", err)
+		return err
+	}
+	return nil
+}
+
+func (s *SkvsStubInterface) GetState(key string) ([]byte, error) {
+	value, found := s.allDataOld[key]
+	if !found {
+		return nil, errors.New("skvs allDataOld key not found")
```

Comment:
```suggestion
		return nil, fmt.Errorf("skvs allDataOld key=%s not found", key)
```

Reply:
Mhh, shouldn't we return `nil, nil` in that case?
```
	// If the key does not exist in the state database, (nil, nil) is returned.
	GetState(key string) ([]byte, error)
```
https://github.com/hyperledger/fabric-chaincode-go/blob/main/shim/interfaces.go#L80C1-L81C38